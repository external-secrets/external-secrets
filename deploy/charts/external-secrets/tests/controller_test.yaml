suite: test controller deployment
templates:
  - deployment.yaml
tests:
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
  - it: should set imagePullPolicy to Always
    set:
      image.pullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should override securityContext
    set:
      podSecurityContext:
        runAsUser: 2000
      securityContext:
        runAsUser: 3000
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsUser: 2000
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            runAsUser: 3000
