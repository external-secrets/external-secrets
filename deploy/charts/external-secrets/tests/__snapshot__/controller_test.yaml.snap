should match snapshot of default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: external-secrets
        app.kubernetes.io/version: v0.8.3
        helm.sh/chart: external-secrets-0.8.3
      name: RELEASE-NAME-external-secrets
      namespace: NAMESPACE
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: external-secrets
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: external-secrets
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - --concurrent=1
              image: ghcr.io/external-secrets/external-secrets:v0.8.3
              imagePullPolicy: IfNotPresent
              name: external-secrets
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
          hostNetwork: false
          serviceAccountName: RELEASE-NAME-external-secrets
